### PR TITLE
Added automated integration tests runs on github (requires approval)

### PR DESCRIPTION
**Pull request by mbfreder**
*Issue #, if available:*

*Description of changes:*

- Added a Github workflow to run integrations tests on every PR. 
- Used environments approval workflow to only run the tests when someone on the team approves it (excluding the person that sent the PR).

Tests:
In order to test, we need to create a dummy PR and verify the following:
- An approval is required to run the workflow
- The workflow successfully runs when someone approves it.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/integ-tests.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  integ-tests:
+    runs-on: ubuntu-latest
+    environment:
+      name: integ-tests
+    steps:
+        - uses: actions/checkout@v3
+        - uses: actions/setup-python@v5
+          with:
+            python-version: '3.11'
+        - name: allows us to build amr64 images
```
Comment (roger-zhangg) [inline .github/workflows/integ-tests.yml:18]: Dose this mean we need to test RIE in both x86 and arm?

Comment (mbfreder) [reply .github/workflows/integ-tests.yml:18]: Yes

**Review by roger-zhangg — APPROVED** [review]
LGTM

File: .github/workflows/integ-tests.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    branches:
+      - develop
+
+jobs:
+  integ-tests:
+    runs-on: ubuntu-latest
+    environment:
+      name: prod
+    steps:
+        - uses: actions/checkout@v3
```
Comment (valerena) [inline .github/workflows/integ-tests.yml:14]: I imagine most of the examples out there use `@v3`, but `@v4` is the [latest version available](https://github.com/actions/checkout), and it avoids this warning that you can see in [your runs](https://github.com/mbfreder/aws-lambda-runtime-interface-emulator/actions/runs/7995969160):

```
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v3. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
```

File: .github/workflows/integ-tests.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        - uses: actions/setup-python@v5
+          with:
+            python-version: '3.11'
+        - name: allows us to build arm64 images
+          run: docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+        - name: run integration tests
+          run: sudo make integ-tests-with-docker
```
Comment (valerena) [inline .github/workflows/integ-tests.yml:21]: do we (or why do we) need to run it with `sudo` ?

Comment (mbfreder) [reply .github/workflows/integ-tests.yml:21]: It doesn't work without it. I ran into some permissions issue when I tried. Even on my dev desktop I have to use it.
```
make integ-tests
make[1]: Entering directory `/local/home/mbfreder/Tooling/aws-lambda-runtime-interface-emulator'
python3 -m venv .venv
Error: [Errno 13] Permission denied: '/local/home/mbfreder/Tooling/aws-lambda-runtime-interface-emulator/.venv/bin/activate.csh'
```

File: .github/workflows/integ-tests.yml (hunk 2 above)
Comment (mbfreder) [reply .github/workflows/integ-tests.yml]: Updated.

**Comment by mbfreder** [comment]
I ran some tests on my fork and can confirm that the workflow doesn't run on the main repo.

**Review by valerena — APPROVED** [review]

**Pull request merged**